core: move files to have a clean kotlin api package

Erashin opened the pull request:
Moved files around to cleanly separate the api files (endpoints, requests, responses, parsers and post processing) from the rest.
Added bonus: corrected a couple of warnings.

bougue-pe commented on core/src/main/kotlin/fr/sncf/osrd/api/RollingStockParser.kt:
```diff
@@ -1,5 +1,6 @@
-package fr.sncf.osrd.api.standalone_sim
+package fr.sncf.osrd.api
 
+import fr.sncf.osrd.api.standalone_sim.PhysicsConsistModel
```
Not sure: Should it move too?

Erashin replied:
Hmm. Honestly there's a couple of objects still used in multiple endpoints that could be cleaned up. Might be worth to create a whole new package all-together for theses objects (maybe call it `model` or something)., For now I'll just leave this as is.

Erashin committed:
core: clean up packages and file locations in kotlin root package

Moved files around to cleanly separate the api files (endpoints, requests, responses, parsers and post processing) from the rest.
Added bonus: corrected a couple of warnings.

Signed-off-by: Erashin <alwenn.charpentier@gmail.com>

bougue-pe reviewed (approved):
Thanks !
(feel free to resolve comment)
